src/extension.ts

```diff
@@ -1319,21 +1329,24 @@ const diffKubernetes = (callback) => {
     getTextForActiveWindow((data, file) => {
         console.log(data, file);
         let kindName = null;
+        let kindObject = null;
         let fileName = null;
 
         let fileFormat = "json";
 
         if (data) {
             fileFormat = (data.trim().length > 0 && data.trim()[0] == '{') ? "json" : "yaml";
-            kindName = findKindNameForText(data);
+            kindObject = findKindNameForText(data);
```

Comment:
Do we need to handle the null case?  What do we do if the object contains a non-default namespace?  (Also applies around line 1348.)

Reply:
Can you provide some pointers on how you would like to see the `null` case handled?

Reply:
I think this would probably be an early exit, possibly with an error message.  `null` implies that we weren't able to recognise the open document as a Kubernetes resource manifest, so we can't diff it; we should probably explain this to the user, but after that there's not much more we can do.  Does that sound right?  Open to feedback!

Reply:
Have added early exit (`return`) with an error message `Cannot Parse Kubernetes Object` to address the `null` case